test/setup/platforms/index.ts

```diff
@@ -35,6 +35,16 @@ const eksSetup: IPlatformSetup = {
   setupTester: eks.setupTester,
 };
 
+// Use a kind cluster pinned to a specific Kubernetes version to mimic OS3.
```

Comment:
why is that?
I mean, I believe you 😛 but maybe there's a doc somewhere about it? how did you learn this?
might be worth documenting in the code around here

Reply:
I think we decided not to try spinning an OpenShift cluster but instead use a regular Kubernetes cluster with the same version that OS3 ships. So that would be 1.11:  https://docs.openshift.com/container-platform/3.11/release_notes/ocp_3_11_release_notes.html#ocp-311-about-this-release
> This release is based on OKD 3.11, and it uses Kubernetes 1.11.

Reply:
not sure I was party to this decision, but all the same - why are we satisfied with that?
that is, when does using KinD give us enough confidence to claim we support OS3?

Reply:
Because OS3 is just a bunch of components built on top of vanilla Kubernetes. More specifically, it's lots of Deployments and CustomResources utilising the Kubernetes API. There aren't any modifications to the core Kubernetes underneath it. So any potential differences would happen at runtime (e.g. some component may decide to block insecure Deployments) and are not at the core platform.

We don't do anything special with our Deployment, and we have verified it on both vanilla Kubernetes and on OpenShift 4. Both of these give us already a fair amount of confidence that it would just work on OS3 as well, because OS3 and OS4 are not very different in terms of behaviour. Also OS3 support is mostly maintenance updates/patches so we really don't expect something major changing in the platform!

Reply:
excellent, thanks for the explanation :)